spnego/spnego.go

```diff
@@ -139,6 +139,15 @@ func (s *SPNEGOToken) Unmarshal(b []byte) error {
 		if err != nil {
 			return fmt.Errorf("not a valid SPNEGO token: %v", err)
 		}
+		// If OID is a KRB5 OID, wrap in a SPNEGO token
+		if oid.Equal(gssapi.OID(gssapi.OIDKRB5)) {
```

Comment:
I think you need to check for both the `OIDKRB5` and the `OIDLegacyKRB5`.  These seem to be interchangeable with Windows Server 2016